Source/WebCore/html/HTMLPlugInImageElement.cpp

```diff
@@ -96,12 +96,15 @@ bool HTMLPlugInImageElement::canLoadURL(const String& relativeURL) const
     return canLoadURL(document().completeURL(relativeURL));
 }
 
-// Note that unlike HTMLFrameElementBase::canLoadURL this uses SecurityOrigin::canAccess.
 bool HTMLPlugInImageElement::canLoadURL(const URL& completeURL) const
 {
     if (completeURL.protocolIsJavaScript()) {
         RefPtr<Document> contentDocument = this->contentDocument();
-        if (contentDocument && !document().securityOrigin().isSameOriginDomain(contentDocument->securityOrigin()))
+        if (!contentDocument) {
+            ASSERT(document().settings().siteIsolationEnabled());
```

Comment:
`LocalFrame` holds a RefPtr to document (so it can be null). So I'm not sure this assert will always be correct.

Reply:
What part of the assertion are you worried about?

Reply:
I am not sure I understand why it matters for this assertion that LocalFrame has a RefPtr to the document.

Reply:
The assertion assumes we return null in `contentDocument()` because `m_contentFrame` is a RemoteFrame (i.e. site isolation is enabled). But it seems like `localFrame->document()` could be null too.

Reply:
This change was incorrect.  There were actually several tests that verified that we can have a null contentDocument without site isolation enabled, and the original version of the PR produced incorrect results.  Updated the PR to check for a RemoteFrame instead.